tests/posix/semaphore/testcase.yaml

```
@@ -0,0 +1,25 @@
+common:
+  filter: not CONFIG_NATIVE_LIBC
+  tags:
+    - posix
+    - semaphore
+  # 1 tier0 platform per supported architecture
+  platform_key:
+    - arch
+    - simulation
+tests:
+  portability.posix.semaphore:
```

Please update so that the test configuration names reflect the option group name (POSIX_SEMAPHORES -> posix.semaphores)
```suggestion
  portability.posix.semaphores:
```